### PR TITLE
fix: use isoWeek and offset weeks correctly on PoP

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: [#12893](https://github.com/lightdash/lightdash/issues/12893)

### Description:

When comparing to previous year, on a week granularity, there was no offset applied to the start of the week, now we address that and values align with the previous year's relative week. 

Also, have used isoWeek plugin so that the it matches the isoString usage we have - thoughts?

Demo:

image

image

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/common/src/utils/metricsExplorer.ts [hunk 1]
```diff
@@ -111,6 +114,18 @@ export const getDateCalcUtils = (timeFrame: TimeFrames) => {
...
             };
         case TimeFrames.YEAR:
+            // Handle week shift for previous year comparison
+            if (grain === TimeFrames.WEEK) {
+                return {
+                    forward: (date: Date) =>
+                        dayjs(date).add(1, 'year').startOf('isoWeek').toDate(),
+                    back: (date: Date) =>
+                        dayjs(date)
+                            .subtract(1, 'year')
```
Comment (joaoviana) [inline packages/common/src/utils/metricsExplorer.ts:124]: subtract, but then shift to start of iso week

File: packages/common/src/utils/metricsExplorer.ts [hunk 2]
```diff
@@ -290,7 +306,7 @@ export const getMetricExplorerDataPointsWithCompare = (
         groupByCompareMetricRows,
         (_, date) =>
             query.comparison === MetricExplorerComparison.PREVIOUS_PERIOD
-                ? getDateCalcUtils(TimeFrames.YEAR)
+                ? getDateCalcUtils(TimeFrames.YEAR, timeFrame)
```
Comment (joaoviana) [inline packages/common/src/utils/metricsExplorer.ts:309]: then, get the forward week with correct start of week

File: packages/frontend/src/features/metricsCatalog/hooks/useDateRangePicker.tsx [hunk 3]
```diff
@@ -367,26 +370,31 @@ export const useDateRangePicker = ({
...
-                                            .startOf('week')
+                                            .startOf('isoWeek')
                                             .toDate() &&
                                     date <=
                                         dayjs(tempDateRange[1])
-                                            .endOf('week')
-                                            .toDate(),
+                                            .endOf('isoWeek')
+                                            .toDate() &&
+                                    !isInFuture, // Don't highlight future dates
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/hooks/useDateRangePicker.tsx:391]: doesnt highlight future days.

File: packages/frontend/src/features/metricsCatalog/hooks/useDateRangePicker.tsx [hunk 4]
```diff
@@ -367,26 +370,31 @@ export const useDateRangePicker = ({
...
                             );
+
                             return {
                                 inRange: isSelected,
-                                firstInRange: date.getDay() === 0 && isSelected, // Highlight the first day of the week
-                                lastInRange: date.getDay() === 6 && isSelected, // Highlight the last day of the week
+                                firstInRange: date.getDay() === 1 && isSelected,
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/hooks/useDateRangePicker.tsx:396]: starts on monday. we should listen to warehouse settings - sunday or monday - but that can be done on a separate PR / in the future

**Review by almeidabbm — APPROVED** [review]

**Pull request merged**